csharp_dotnetcore/Startup.cs

```diff
@@ -0,0 +1,130 @@
+                }
+
+                options.CredentialProvider = new SimpleCredentialProvider(endpointService.AppId, endpointService.AppPassword);
+
+                // Creates a logger for the application to use.
+                ILogger logger = _loggerFactory.CreateLogger<FoodBot>();
+
+                // Catches any errors that occur during a conversation turn and logs them.
+                options.OnTurnError = async (context, exception) =>
+                {
+                    logger.LogError($"Exception caught : {exception}");
+                    await context.SendActivityAsync("Sorry, it looks like something went wrong.");
+                };
+            });
+
+            // The Memory Storage used here is for local bot debugging only. When the bot
+            // is restarted, everything stored in memory will be gone.
+            IStorage dataStore = new MemoryStorage();
+
+            // For production bots use the Azure Blob or
+            // Azure CosmosDB storage providers. For the Azure
+            // based storage providers, add the Microsoft.Bot.Builder.Azure
+            // Nuget package to your solution. That package is found at:
+            // https://www.nuget.org/packages/Microsoft.Bot.Builder.Azure/
+            // Uncomment the following lines to use Azure Blob Storage
+            // //Storage configuration name or ID from the .bot file.
+            // const string StorageConfigurationId = "<STORAGE-NAME-OR-ID-FROM-BOT-FILE>";
+            // var blobConfig = botConfig.FindServiceByNameOrId(StorageConfigurationId);
+            // if (!(blobConfig is BlobStorageService blobStorageConfig))
+            // {
+            //    throw new InvalidOperationException($"The .bot file does not contain an blob storage with name '{StorageConfigurationId}'.");
+            // }
+            // // Default container name.
+            // const string DefaultBotContainer = "<DEFAULT-CONTAINER>";
+            // var storageContainer = string.IsNullOrWhiteSpace(blobStorageConfig.Container) ? DefaultBotContainer : blobStorageConfig.Container;
+            // IStorage dataStore = new Microsoft.Bot.Builder.Azure.AzureBlobStorage(blobStorageConfig.ConnectionString, storageContainer);
+
+            // Create Conversation State object.
+            // The Conversation State object is where we persist anything at the conversation-scope.
+            var conversationState = new ConversationState(dataStore);
+
+            // Register conversation state.
+            services.AddSingleton<BotState>(conversationState);
+            // The dialogs will need a state store accessor. Creating it here once (on-demand) in the ConversationState allows the dependency injection
+            // to hand it to our top-level DialogSet class, FoodBotDialogSet, that is create per-request.
+            services.AddSingleton<IStatePropertyAccessor<DialogState>>(conversationState.CreateProperty<DialogState>("FOODBOTDIALOGSTATE"));
```

Comment:
`So there's two things here.

 1. If, after reading `2`, you end up keeping the singleton registration for the `IStatePropertyAccessor<DialogState>`, I would simplify it to just be:

```CSharp
services.AddSingleton(conversationState.CreateProperty<DialogState>("FOODBOTDIALOGSTATE"));
```

The generic type will be resolved by the return type, so there's no need to specify it explicitly.

 2. Rather than registering the singleton `IStatePropertyAccessor<DialogState>` at all since it's only going to be used by the `FoodBotDialogSet`, you can just use an overload of `AddSingleton` that allows you to inline create the dialog set yourself and supply the property accessor to the constructor manually.

```CSharp
services.AddSingleton(sp => new FoodBotDialogSet(conversationState.CreateProperty("FOODBOTDIALOGSTATE")));
```

If you wanted to get even fancier, since you're now registering the `ConversationState`, you could clean up the whole thing to just be:

```CSharp
 // Register conversation state. 
services.AddSingleton<BotState>(new ConversationState(storage));
services.AddSingleton(sp => new FoodBotDialogSet(sp.GetRequiredService<BotState>().CreateProperty("FOODBOTDIALOGSTATE")));
```